rclcpp/test/rclcpp/test_logger.cpp

```diff
@@ -210,15 +210,7 @@ TEST(TestLogger, get_logging_directory) {
   ASSERT_EQ(true, rcutils_set_env("ROS_LOG_DIR", nullptr));
   ASSERT_EQ(true, rcutils_set_env("ROS_HOME", nullptr));
 
-  auto path = rclcpp::get_logging_directory();
+  auto path = rclcpp::get_log_directory();
   auto expected_path = rcpputils::fs::path{"/fake_home_dir"} / ".ros" / "log";
-
-  // TODO(ivanpauno): Add operator== to rcpputils::fs::path
-  auto it = path.cbegin();
-  auto eit = expected_path.cbegin();
-  for (; it != path.cend() && eit != expected_path.cend(); ++it, ++eit) {
-    EXPECT_EQ(*eit, *it);
-  }
-  EXPECT_EQ(it, path.cend());
-  EXPECT_EQ(eit, expected_path.cend());
+  EXPECT_EQ(path.string(), expected_path.string());
```

Comment:
You should be able to just do

```suggestion
  EXPECT_EQ(path, expected_path);
```

Reply:
And I know that the previous code just compared the strings, but should we instead check for equivalence using [`std::filesystem::equivalent`](https://en.cppreference.com/w/cpp/filesystem/equivalent), e.g., in case there are symlinks or anything like that? It might not matter for a test, but I thought I'd mention it.